BasiliskII/src/rom_patches.cpp

```diff
@@ -1008,6 +1008,179 @@ static bool patch_rom_classic(void)
 	*wp++ = htons(M68K_EMUL_OP_IRQ);
 	*wp++ = htons(0x4a80);		// tst.l	d0
 	*wp = htons(0x67f4);		// beq		0x402be2
+
+
+	// Patch guest screen for an arbitary resolution
+	// We are going to steal the ROM space that runs RAM test
+	uint32 patchCodeBaseMac; 	// a stateful variable to keep track of the stolen location of the patched code in guest memory.
+	uint16* patchwp; 					// a stateful word pointer pointing to the patch code in host memory.
+	uint32* patchlp;					// a stateful long pointer pointing to the patch code in host memory.
+
+	uint32* lp;								// a statless long pointer
+
+	wp = (uint16 *)(ROMBaseHost + 0x5ca); // start to patch P_mInitVideoGlobal route
+	*wp++ = htons(0x4eb9); /* JSR */
+	lp = (uint32 *)wp;
+
+	const uint32 patchCodeOffset = 0x1d3e; // we steal P_mRamTest routine space
+	patchCodeBaseMac = ROMBaseMac + patchCodeOffset;
+	uint8* patchCodeBaseHost = ROMBaseHost + patchCodeOffset;
+	*lp++ = htonl(patchCodeBaseMac); // we JSR to this Mac address
+	wp = (uint16 *)lp;
+
+	patchwp = (uint16 *)(patchCodeBaseHost); // the host address for the patch
+	// MOVE.L $MacFrameBaseMac24Bit, ($ScrnBase)
+	*patchwp++ = htons(0x21fc); /* MOVE.L */
+	patchCodeBaseMac += 2;
+
+	patchlp = (uint32 *)patchwp;
+	*patchlp++ = htonl(MacFrameBaseMac24Bit);
+	patchCodeBaseMac += 4;
+
+	patchwp = (uint16 *)patchlp;
+	*patchwp++ = htons(0x0824); /* (ScrnBase) */
+	patchCodeBaseMac += 2;
+
+	*patchwp++ = htons(0x4e75); /* RTS */
+	patchCodeBaseMac += 2;
+	patchlp = (uint32 *)patchwp; // keep in sync
+
+	// continue to patch P_mInitVideoGlobal routine
+	wp = (uint16 *)(ROMBaseHost + 0x5d2);
+	*wp = htons(MacScreenWidth / 8);
+	wp = (uint16 *)(ROMBaseHost + 0x60a);
+	*wp = htons(MacScreenHeight);
+	wp = (uint16 *)(ROMBaseHost + 0x60e);
+	*wp = htons(MacScreenWidth);
+
+	wp = (uint16 *)(ROMBaseHost + 0x8cc);
+	*wp = htons(MacScreenHeight);
+	wp = (uint16 *)(ROMBaseHost + 0x8ce);
+	*wp = htons(MacScreenWidth);
+
+	// blink floppy, disk icon
+	lp = (uint32 *)(ROMBaseHost + 0xf4c);
+	*lp = htonl(MacFrameBaseMac24Bit
+							+ (((MacScreenHeight / 4) * 2 - 25) * MacScreenWidth
```

Comment:
Nit: This formatting is pretty hard to follow. Can you format it differently?

Alternatively, perhaps you can make a helper function that can be re-used in the different places below? The parameters could then also be named according to their meaning.

Reply:
![rom_patches cpp](https://user-images.githubusercontent.com/3475933/87617015-7d954880-c6e4-11ea-8add-e52f04f65eb2.png)

They are all tabs. See the image "->".

In Qt Creator, I set the visualization two spaces equals to one tab. Thus, you see the alignment. Unless you want to convert the tab into spaces, I don't see any better solution.

Reply:
Given the fact that we have wider screen, I just unwrapped the formulas. 
Done.